securityscanutils/securityscan.go

```diff
@@ -0,0 +1,281 @@
+		}
+		// Filter releases by version constraint provided
+		filteredReleases := githubutils.FilterReleases(allReleases, opts.VersionConstraint)
+		githubutils.SortReleasesBySemver(filteredReleases)
+		for _, release := range filteredReleases {
+			// We can swallow the error here, any releases with improper tag names
+			// will not be included in the filtered list
+			ver, _ := semver.NewVersion(release.GetTagName())
+			err = repo.RunMarkdownScan(ver, markdownTplFile)
+			if err != nil {
+				return eris.Wrapf(err, "error generating markdown file from security scan for version %s", release.GetTagName())
+			}
+			// Only generate sarif files if we are uploading code scan results
+			// to github
+			if repo.Opts.UploadCodeScanToGithub {
+				err = repo.RunGithubSarifScan(ver, sarifTplFile)
+				if err != nil {
+					return eris.Wrapf(err, "error generating github sarif file from security scan for version %s", release.GetTagName())
+				}
+			}
+		}
+
+	}
+	return nil
+}
+
+func (r *SecurityScanRepo) RunMarkdownScan(versionToScan *semver.Version, markdownTplFile string) error {
+	images, err := r.GetImagesToScan(versionToScan)
+	if err != nil {
+		return err
+	}
+	version := versionToScan.String()
+	outputDir := path.Join(r.Opts.OutputDir, r.Repo, "markdown_results", version)
+	err = osutils.CreateDirIfNotExists(outputDir)
+	if err != nil {
+		return err
+	}
+	for _, image := range images {
+		imageWithRepo := fmt.Sprintf("%s/%s:%s", r.Opts.ImageRepo, image, version)
+		fileName := fmt.Sprintf("%s_cve_report.docgen", image)
+		output := path.Join(outputDir, fileName)
+		_, err = RunTrivyScan(imageWithRepo, version, markdownTplFile, output)
+		if err != nil {
+			return eris.Wrapf(err, "error running image scan on image %s", imageWithRepo)
+		}
+	}
+	return nil
+}
+
+func (r *SecurityScanRepo) RunGithubSarifScan(versionToScan *semver.Version, sarifTplFile string) error {
+	images, err := r.GetImagesToScan(versionToScan)
+	if err != nil {
+		return err
+	}
+	version := versionToScan.String()
+	outputDir := path.Join(r.Opts.OutputDir, r.Repo, "sarif_results", version)
+	err = osutils.CreateDirIfNotExists(outputDir)
+	if err != nil {
+		return err
+	}
+	for _, image := range images {
+		imageWithRepo := fmt.Sprintf("%s/%s:%s", r.Opts.ImageRepo, image, version)
+		fileName := fmt.Sprintf("%s_cve_report.sarif", image)
+		output := path.Join(outputDir, fileName)
+		success, err := RunTrivyScan(imageWithRepo, version, sarifTplFile, output)
+		if err != nil {
+			return eris.Wrapf(err, "error running image scan on image %s", imageWithRepo)
+		}
+		if success {
+			err = r.UploadSecurityScanToGithub(output, version)
+			if err != nil {
+				return eris.Wrapf(err, "error uploading security scan results sarif to github for version %s", version)
+			}
+		}
+	}
+	return nil
+}
+
+func (r *SecurityScanRepo) GetImagesToScan(versionToScan *semver.Version) ([]string, error) {
```

Comment:
commented above, but wondering what the difference between this function is, and the github_utils.FilterReleases is